src/encoded/utils.py

```diff
@@ -92,3 +99,102 @@ def get_configuration_value(property_name: str,
             if raise_exception is True:
                 raise Exception(f"Cannot get configuration value for: {property_name}\n{get_error_message(e)}")
     return fallback
+
+
+def get_item_with_testapp(
+    testapp: TestApp,
+    identifier: str,
+    collection: Optional[str] = None,
+    frame: Optional[str] = None,
+    status: Optional[Union[int, List[int]]] = None,
+) -> Dict[str, Any]:
+    """Get item from test app."""
+    add_on = get_frame_add_on(frame)
+    resource_path = get_formatted_resource_path(
+        identifier, collection=collection, add_on=add_on
+    )
+    response = testapp.get(resource_path, status=status)
+    if response.status_int == 301:
+        return response.follow().json
+    return response.json
+
+
+def get_frame_add_on(frame: Union[str, None]) -> str:
+    """Format frame parameter, if provided."""
+    if frame:
+        return f"frame={frame}"
+    return ""
+
+
+def get_formatted_resource_path(
+    identifier: str, collection: Optional[str] = None, add_on: Optional[str] = None
+) -> str:
+    """Format resource path for URL expectations."""
+    resource_path_with_add_on = get_resource_path_with_add_on(
+        identifier, collection, add_on
+    )
+    if not resource_path_with_add_on.startswith("/"):
+        return f"/{resource_path_with_add_on}"
+    return resource_path_with_add_on
+
+
+def get_resource_path_with_add_on(
+    identifier: str, collection: Optional[str] = None, add_on: Optional[str] = None
+) -> str:
+    """Get resource path with optional URL parameters, if provided."""
+    resource_path = get_resource_path(identifier, collection)
+    if add_on:
+        return f"{resource_path}?{add_on}"
+    return resource_path
+
+
+def get_resource_path(identifier: str, collection: Optional[str] = None) -> str:
+    """Get resource path with collection, if provided."""
+    if collection:
+        return get_formatted_collection_resource_path(identifier, collection)
+    return identifier
+
+
+def get_formatted_collection_resource_path(identifier: str, collection: str) -> str:
+    """Format to '{collection}/{identifier}/'.
+
+    Collection can be snake- or camel-cased.
+    """
+    dashed_collection = get_kebab_formatted_collection(collection)
+    return f"/{pluralize_collection(dashed_collection)}/{identifier}/"
+
+
+def get_kebab_formatted_collection(collection: str) -> str:
+    """Format collection to  kebab case.
+
+    Intended for collection to come in camel- or snake-case.
+    """
+    return to_snake_case(collection).replace("_", "-")
+
+
+def pluralize_collection(collection: str) -> str:
```

Comment:
Does this method not work? https://github.com/4dn-dcic/utils/blob/master/dcicutils/lang_utils.py#L114